### ISSUE TITLE
HASURA_GRAPHQL_CORS_DOMAIN does not want to apply in docker-compose file

### ISSUE THREAD

**Issue by lemoinepierre** [hasura/graphql-engine#3511]
Hey Hasura,
I am trying to securize my installastion and after setting up admin secret, I would like to use **HASURA_GRAPHQL_CORS_DOMAIN** and **HASURA_GRAPHQL_AUTH_WEBHOOK**.
I am using docker v1.0.0.-rc1 on linux and here is my docker-file

```
version: '3.3'
services:
  postgres:
    image: postgres
    restart: always
    volumes:
    - db_data:/var/lib/postgresql/data
  graphql-engine:
    image: hasura/graphql-engine:v1.0.0-rc.1
    ports:
    - "8080:8080"
    depends_on:
    - "postgres"
    restart: always
    environment:
      HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:@postgres:5432/postgres
      #      HASURA_GRAPHQL_ADMIN_SECRET: #######################
      HASURA_GRAPHQL_CORS_DOMAIN: "http://localhost:8080"
      HASURA_GRAPHQL_ENABLE_CONSOLE: "true" # set to "false" to disable console
      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
      ## uncomment next line to set an admin secret
      HASURA_GRAPHQL_AUTH_WEBHOOK: "http://3c13e9cd9eea:5000/auth-webhook"
volumes:
  db_data:
```
This docker is running on a linux VM that I am accessing from my windows PC.
I can access it either via an SSH tunnel or directly with 192.168.56.104:8080

1. I am able to access with both address 192.168.56.104:8080 http://localhost:8080, no matter if I put HASURA_GRAPHQL_CORS_DOMAIN or not
2. When I do access this link http://192.168.56.104:8080/v1alpha1/config. I see this
 > {"version":"v1.0.0-rc.1","is_admin_secret_set":false,"is_auth_hook_set":false,"is_jwt_set":false,"jwt":null}

So in summary, Hasura does not see HASURA_GRAPHQL_CORS_DOMAIN HASURA_GRAPHQL_AUTH_WEBHOOK. However, It sees HASURA_GRAPHQL_ENABLE_CONSOLE as if I put false, I don't have access to the console anymore.

Do you have any idea where it's coming from?

P.S.: Hasura is a fantastic tool and very easy to use. Well done to the entire team.

**Comment by shahidhk** [comment]
> I am able to access with both address 192.168.56.104:8080 http://localhost:8080, no matter if I put HASURA_GRAPHQL_CORS_DOMAIN or not

Setting CORS domain will only block cross-origin API requests to Hasura. It will not stop you from serving the API or the console on a different IP adress. Only when that particular console makes a request to the API it should be blocked if it is not from localhost:8080

> When I do access this link http://192.168.56.104:8080/v1alpha1/config. I see this

Yes, opening this address on the browser is not considered as a cross origin request. See https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS for more details.

**Comment by lemoinepierre** [comment]
I did not known, thanks.
